### ISSUE TITLE
[11] Calendar: Start date cannot be changed in form view → Error: Failed to render template

### ISSUE THREAD

**Issue by bosskopp** [odoo/odoo#20238]
**Impacted versions:**
11 community & enterprise
Also tested on https://demo.odoo.com

**Steps to reproduce:**

1. Goto "Calendar" app
2. Create new event and set start date to now (not recurring, not "all day")
3. Save event
4. Change event start and try to save again

Note: Changing the start-date via drag and drop in the calendar-view works!

**Current behavior:**

Warning:
`Failed to render template <Template memory:7f02ff464780> using values {'user': res.users(1,), 'format_date': <function MailTemplate.render_template.<locals>.<lambda> at 0x7f0309624268>, 'format_tz': <function MailTemplate.render_template.<locals>.<lambda> at 0x7f0309624598>, 'object': calendar.attendee(24,), 'format_amount': <function MailTemplate.render_template.<locals>.<lambda> at 0x7f02ff467378>, 'ctx': {'tz': 'Europe/Brussels', 'dbname': 'demo_110_1508223013', 'uid': 1, 'action_id': 144, 'base_url': 'https://demo3.odoo.com', 'color': {'accepted': 'green', 'declined': 'red', 'tentative': '#FFFF00', 'needsAction': 'grey'}, 'lang': 'en_US', 'safe': False}}
`

**Expected behavior:**

Event date changed.

**Video/Screenshot:**
![odoo11-cal-err](https://user-images.githubusercontent.com/1733427/31656556-2cc45956-b32c-11e7-969e-cfc5dd06852b.gif)

**Comment by bosskopp** [comment]
```
INFO dev odoo.addons.mail.models.mail_template: Failed to render template <Template memory:7f44d15f8860> using values {'format_tz': <function MailTemplate.render_template.<locals>.<lambda> at 0x7f44d05560d0>, 'format_amount': <function MailTemplate.render_template.<locals>.<lambda> at 0x7f44d05562f0>, 'object': calendar.attendee(22,), 'format_date': <function MailTemplate.render_template.<locals>.<lambda> at 0x7f44f0295048>, 'user': res.users(1,), 'ctx': {'color': {'tentative': '#FFFF00', 'accepted': 'green', 'needsAction': 'grey', 'declined': 'red'}, 'lang': 'de_DE', 'base_url': 'http://127.0.0.1:8069', 'uid': 1, 'params': {'menu_id': 86, 'action': 114, '_push_me': False}, 'action_id': 114, 'dbname': 'dev', 'tz': 'Europe/Berlin', 'safe': False}}
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/addons/mail/models/mail_template.py", line 371, in render_template
    render_result = template.render(variables)
  File "/usr/local/lib/python3.5/dist-packages/jinja2/environment.py", line 989, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/lib/python3.5/dist-packages/jinja2/environment.py", line 754, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.5/dist-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "<template>", line 35, in top-level template code
  File "/usr/local/lib/python3.5/dist-packages/jinja2/sandbox.py", line 355, in call
    return __context.call(__obj, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/calendar/models/calendar.py", line 1278, in get_display_time_tz
    return self._get_display_time(self.start, self.stop, self.duration, self.allday)
  File "/usr/lib/python3/dist-packages/odoo/addons/calendar/models/calendar.py", line 686, in _get_display_time
    date_str = date.strftime(format_date).decode('utf-8')
AttributeError: 'str' object has no attribute 'decode'
```

**Comment by Yenthe666** [comment]
Hi @bosskopp,

Thanks a lot for your bug report - I can confirm this is a bug.
@JKE-be or @nim-odoo could you guys ook at it?

Regards,
Yenthe

**Comment by Elkasitu** [comment]
Hi, I believe this has been fixed in rev f5df4fb
It was an issue with py2-py3 compatibility

@Yenthe666 can you confirm and close? Thanks

**Comment by Yenthe666** [comment]
Thank you for the update @Elkasitu  - yes the commit https://github.com/odoo/odoo/commit/f5df4fbca1d6dce4f98cdb6b27bee3fb843febf4 fixed the bug!
